pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/SierraIdentifiers.scala

```diff
@@ -7,41 +7,44 @@ import weco.catalogue.internal_model.identifiers.{
 import weco.sierra.models.SierraQueryOps
 import weco.sierra.models.data.SierraBibData
 import weco.sierra.models.identifiers.SierraBibNumber
+import weco.sierra.models.marc.Subfield
+
+import scala.util.matching.Regex
 
-// Populate wwork:identifiers.
-//
-//    We populate with the following identifiers:
-//
-//    -   "sierra-identifier" for the 7-digit internal ID
-//
-//    -   "isbn" from MARC tag 020 subfield $a.  This is repeatable.
-//        https://www.loc.gov/marc/bibliographic/bd020.html
-//
-//    -   "issn" from MARC tag 022 ǂa.  This is repeatable.
-//        https://www.loc.gov/marc/bibliographic/bd022.html
-//
-//    -   "wellcome-digcode" from MARC tag 759 ǂa.  This is repeatable.
-//        Note: MARC 759 is not assigned by the Library of Congress.
-//
 object SierraIdentifiers
     extends SierraIdentifiedDataTransformer
     with SierraQueryOps {
 
   type Output = List[SourceIdentifier]
 
   def apply(bibId: SierraBibNumber,
-            bibData: SierraBibData): List[SourceIdentifier] = {
-    val sierraIdentifier = SourceIdentifier(
-      identifierType = IdentifierType.SierraIdentifier,
-      ontologyType = "Work",
-      value = bibId.withoutCheckDigit
-    )
+            bibData: SierraBibData): List[SourceIdentifier] =
+    createSierraIdentifier(bibId) ++
+      getIsbnIdentifiers(bibData) ++
+      getIssnIdentifiers(bibData) ++
+      getDigcodes(bibData) ++
+      getIconographicNumbers(bibData) ++
+      getEstcReferences(bibData)
 
-    List(sierraIdentifier) ++ getIsbnIdentifiers(bibData) ++ getIssnIdentifiers(
-      bibData) ++ getDigcodes(bibData) ++ getIconographicNumbers(bibData)
-  }
+  /** Create a seven-digit ID based on the internal ID.
+    *
+    * We use the eight-digit ID with check digit as the sourceIdentifier on the Work.
+    *
+    */
+  private def createSierraIdentifier(
```

Comment:
bikeshed: if it returns a `List[SourceIdentifier]` it is arguably a `createSierraIdentifiers`